### DISCUSSION TITLE
What about using MLJAR for  time-series prediction? Are there pitfalls?

### DISCUSSION THREAD

**Discussion by madpower2000** [mljar/mljar-supervised#493] [Q&A]
What about using MLJAR for  time-series prediction? Are there pitfalls?

**Comment by pplonski** [comment]
Hi @madpower2000,

The MLJAR AutoML can be used for time-series prediction if you convert your data into stationary data. The AutoML can't itself do feature engineering to convert time-series data into a stationary problem. It can be implemented. I'm open to suggestions for new feature engineering methods and new algorithms that are suitable for time-series data.

**Comment by madpower2000** [comment]
Hi @pplonski,
From my point of view, feature preprocessing is not the problem at all and easily can be done by two (or more) lines of code in python outside AutoML, besides it’s domain specific and "one size fits all" approach doesn't work here.

My concern is about cross validation under the hood of Auto ML. For time series data with temporal dependence must be done with some different technique than for just tabular data. Take a look to sklearn.model_selection.TimeSeriesSplit() documentation.

In depth topic discussion can be found here:

- “A note on the validity of cross-validation for evaluating autoregressive time series prediction” by Rob J Hyndman (https://robjhyndman.com/publications/cv-time-series/
- Aileen Nielsen. “Practical Time Series Analysis Prediction with Statistics & Machine Learning”. O’Reilly.2020

**Reply by pplonski** [reply]
There is an option to set custom validation in AutoML. Should be easy to use with TimeSeriesSplit. For custom validation details please check https://github.com/mljar/mljar-supervised/issues/401

Please let me know if it works for you. If you need help I can create example Python code.

**Reply by madpower2000** [reply]
Thanks, I'll check it out.
